src/core/utils/history_tracker.cpp

```diff
@@ -469,19 +469,19 @@ void HistoryTracker::EntryAgeToString(uint32_t aEntryAge, char *aBuffer, uint16_
 
     if (aEntryAge >= kMaxAge)
     {
-        writer.Append("more than %u days", kMaxAge / Time::kOneDayInMsec);
+        writer.Append("more than %lu days", kMaxAge / Time::kOneDayInMsec);
     }
     else
     {
         uint32_t days = aEntryAge / Time::kOneDayInMsec;
 
         if (days > 0)
         {
-            writer.Append("%u day%s ", days, (days == 1) ? "" : "s");
+            writer.Append("%lu day%s ", days, (days == 1) ? "" : "s");
             aEntryAge -= days * Time::kOneDayInMsec;
         }
 
-        writer.Append("%02u:%02u:%02u.%03u", (aEntryAge / Time::kOneHourInMsec),
+        writer.Append("%02lu:%02lu:%02lu.%03lu", (aEntryAge / Time::kOneHourInMsec),
```

Comment:
Suggest keeping the `%02u` as is.
Instead we can add `static_cast<uint16_t>()` to all values (since these are minute and seconds in hour).